solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts

```diff
@@ -1254,41 +1260,92 @@ describe("RandomBeacon - Group Creation", () => {
-              await createSnapshot()
-              ;({
-                transaction: tx,
-                dkgResult,
-                dkgResultHash,
-              } = await signAndSubmitCorrectDkgResult(
-                randomBeacon,
-                groupPublicKey,
-                genesisSeed,
-                startBlock,
-                misbehavedIndices
-              ))
-            })
+                before(async () => {
+                  await createSnapshot()
+                  ;({
+                    transaction: tx,
+                    dkgResult,
+                    dkgResultHash,
+                  } = await signAndSubmitCorrectDkgResult(
+                    randomBeacon,
+                    groupPublicKey,
+                    genesisSeed,
+                    startBlock,
+                    misbehavedIndices
+                  ))
+                })
 
-            after(async () => {
-              await restoreSnapshot()
-            })
+                after(async () => {
+                  await restoreSnapshot()
+                })
 
-            it("should succeed with misbehaved members", async () => {
-              await expect(tx)
-                .to.emit(randomBeacon, "DkgResultSubmitted")
-                .withArgs(
-                  dkgResultHash,
-                  genesisSeed,
-                  dkgResult.submitterMemberIndex,
-                  dkgResult.groupPubKey,
-                  dkgResult.misbehavedMembersIndices,
-                  dkgResult.signatures,
-                  dkgResult.signingMembersIndices,
-                  dkgResult.members
-                )
-            })
+                it("should succeed with misbehaved members", async () => {
+                  await expect(tx)
+                    .to.emit(randomBeacon, "DkgResultSubmitted")
+                    .withArgs(
+                      dkgResultHash,
+                      genesisSeed,
+                      dkgResult.submitterMemberIndex,
+                      dkgResult.groupPubKey,
+                      dkgResult.misbehavedMembersIndices,
+                      dkgResult.signatures,
+                      dkgResult.signingMembersIndices,
+                      dkgResult.members
+                    )
+                })
+
+                it("should correctly set a group members hash", async () => {
+                  const storedGroup = await randomBeacon["getGroup(bytes)"](
+                    groupPublicKey
+                  )
+
+                  const expectedMembers = [...dkgResult.members]
+                  expectedMembers.splice(1, 1) // index -1
+                  expectedMembers.splice(7, 1) // index -2 (cause expectedMembers already shrinked)
+                  expectedMembers.splice(8, 1) // index -3
+                  expectedMembers.splice(26, 1) // index -4
+                  expectedMembers.splice(55, 1) // index -5
+                  expectedMembers.splice(58, 1) // index -6
```

Comment:
We could add a comment above that says misbehaved indices are `[2, 9, 11, 30, 60, 64]`. This could help to understand the magic happening here.

Reply:
e6ba9ef